airbyte-cdk/python/airbyte_cdk/utils/traced_exception.py

```diff
@@ -61,18 +67,18 @@ def as_airbyte_message(self, stream_descriptor: StreamDescriptor = None) -> Airb
                 internal_message=self.internal_message,
                 failure_type=self.failure_type,
                 stack_trace=stack_trace_str,
-                stream_descriptor=stream_descriptor,
+                stream_descriptor=self._stream_descriptor if self._stream_descriptor is not None else stream_descriptor,
             ),
         )
 
         return AirbyteMessage(type=MessageType.TRACE, trace=trace_message)
 
-    def as_connection_status_message(self) -> AirbyteMessage:
+    def as_connection_status_message(self) -> Optional[AirbyteMessage]:
```

Comment:
Was this an intentional change for this PR? we were talking a lot about traced exception descriptors, but this is related to making check status messages optional. The changes seem okay, but I'm missing a little context into this change. Was there another Salesforce bug that this was needed for?

Reply:
This was a typing issue, I didn't actually change the behavior. Let me know if I accidentally did though and I'll fix this